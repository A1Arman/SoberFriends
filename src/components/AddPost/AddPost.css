.addPost-grid-container {
    display: grid;
    grid-template-areas: "label ..."
                         "input input"
                         "label2 ..."
                         "textarea textarea";
    margin: 0 3rem;
}


.sub-btn, .rst-btn {
    margin: 0 1rem;
    border-radius: 5px;
}


.addPost-label {
    grid-area: label;
    text-align: start;
    margin: 1rem 0;
    color: #265573;
    font-weight: bold;
    font-size: 1.75rem;
}

.sub-btn span, .rst-btn span {
    cursor: pointer;
    display: inline-block;
    position: relative;
    transition: 0.5s;
}
  
.sub-btn span::after, .rst-btn span::after {
    content: '\00bb';
    position: absolute;
    opacity: 0;
    top: 0;
    right: -20px;
    transition: 0.5s;
}
  
.sub-btn:hover span, .rst-btn:hover span {
    padding-right: 25px;
}
  
.sub-btn:hover span::after, .rst-btn:hover span::after {
    opacity: 1;
    right: 0;
}

.sub-btn, .rst-btn {
    border: none;
    color: white;
    cursor: pointer;
    font-weight: bold;
    outline: none;
    padding: 1rem;
    transition: all 0.5s;
}

.sub-btn {
    background-color: #265573;
}

.rst-btn {
    background-color: #265573;
}

#post_title {
    width: 100%;
    height: 3rem;
    border-radius: 4px;
    background-color: #f8f8f8;
    grid-area: input;
}

#post_content {
    width: 100%;
    height: 15rem;
    padding: 12px 20px;
    box-sizing: border-box;
    border: 2px solid #ccc;
    border-radius: 4px;
    background-color: #f8f8f8;
    resize: none;
    margin-bottom: 1rem;
    grid-area: textarea;
  }


  @media screen and (min-width: 992px) {
      .addPost-grid-container {
          margin: 0 15rem;
      }
  }