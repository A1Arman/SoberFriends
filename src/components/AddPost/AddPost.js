import React from 'react';
import './AddPost.css';

function AddPost(props) {
    return (
        <main role="main">
            <header>
                <h1>New Post</h1>
            </header>
            <section>
                <form id="add-post-form" onSubmit={props.handleSubmit}>
                <div className="form-section grid-container">
                    <label className='addPost-label' htmlFor="post_title">Post title</label>
                    <input type="text" name="post_title" id="post_title" placeholder="Feeling Unstoppable" required />
                </div>
                <div className="form-section grid-container">
                    <label className='addPost-label' htmlFor="post_content">Post Content<br/>
                        (Tell us about your current journey, how you are feeling, or just any tips you might have!)</label>
                    <textarea name="post_content" rows="10" id="post_content" required></textarea>
                </div>
                <button type="submit" className='sub-btn' onClick={props.handleValidation}><span>Submit</span></button>
                <button type="reset" className='rst-btn'><span>Reset</span></button>
                {props.contentValid ? null : <p>Both title and content field are required</p>}
                </form>
            </section>
        </main>
    )
}

export default AddPost;