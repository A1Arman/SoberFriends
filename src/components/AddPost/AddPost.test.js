import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import AddPost from './AddPost';


describe(`LoginForm component`, () => {
    it('renders without crashing', () => {
        const wrapper = shallow(<AddPost />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});