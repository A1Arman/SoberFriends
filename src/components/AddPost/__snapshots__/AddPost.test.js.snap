// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm component renders without crashing 1`] = `
<main
  role="main"
>
  <header>
    <h1>
      New Post
    </h1>
  </header>
  <section>
    <form
      id="add-post-form"
    >
      <div
        className="form-section grid-container"
      >
        <label
          htmlFor="post_title"
        >
          Post title
        </label>
        <input
          id="post_title"
          name="post_title"
          placeholder="Feeling Unstoppable"
          required={true}
          type="text"
        />
      </div>
      <div
        className="form-section grid-container"
      >
        <label
          htmlFor="post_content"
        >
          Post Content
          <br />
          (Tell us about your current journey, how you are feeling, or just any tips you might have!)
        </label>
        <textarea
          id="post_content"
          name="post_content"
          required={true}
          rows="10"
        />
      </div>
      <button
        className="sub-btn"
        type="submit"
      >
        Submit
      </button>
      <button
        className="rst-btn"
        type="reset"
      >
        Reset
      </button>
      <p>
        Both title and content field are required
      </p>
    </form>
  </section>
</main>
`;
